assets/src/adaptivity/scripting.ts

```diff
@@ -57,25 +57,15 @@ export const getExpressionStringForValue = (
 
     // it might be CSS string, which can be decieving
     let actuallyAString = false;
+    const expressions = extractAllExpressionsFromText(val);
+    // A expression will not have a ';' inside it. So if there is a ';' inside it, it is CSS.
+    const isCSSString = expressions.filter((e) => e.includes(';'));
+    if (isCSSString?.length) {
+      actuallyAString = true;
+    }
     try {
       const testEnv = new Environment(env);
-      const evalResult = evalScript(`let foo = ${val};`, testEnv);
-      // when evalScript is executed successfully, evalResult.result is null.
-      // evalScript does not trigger catch block even though there is error and add the error in stack property.
-      if (evalResult?.result !== null) {
-        try {
-          //trying to check if it is a CSS string.This might not handle any advance CSS string.
-          const matchingCssElements = val.match(
-            /^(([a-z0-9\\[\]=:]+\s?)|((div|span|body.*|.box-sizing:*|.columns-container.*|background-color.*)?(#|\.){1}[a-z0-9\-_\s?:]+\s?)+)(\{[\s\S][^}]*})$/im,
-          );
-          //matchingCssElements !== null then it means it's a CSS string so set actuallyAString=true so that it can be wrapped in ""
-          if (matchingCssElements) {
-            actuallyAString = true;
-          }
-        } catch (e) {
-          actuallyAString = true;
-        }
-      }
+      evalScript(`let foo = ${val};`, testEnv);
```

Comment:
still we need to test `evalResult != null` for things besides CSS. also I think if we have determined via the CSS test above that `if (actuallyAString === true)` then we don't need to continue testing

Reply:
Added changes for the second point. I am still not sure why we need the evalResult != null section because if it fails we already setting  "actuallyAString = true;" in catch section.